mesonbuild/compilers/mixins/visualstudio.py

```diff
@@ -437,6 +437,24 @@ def get_instruction_set_args(self, instruction_set: str) -> T.Optional[T.List[st
     def get_pch_base_name(self, header: str) -> str:
         return os.path.basename(header)
 
+    # Linking ASM-only objects into an executable or DLL
+    # require this, otherwise it'll fail to find
+    # _WinMain or _DllMainCRTStartup.
+    def get_crt_link_args(self, crt_val: str, buildtype: str) -> T.List[str]:
```

Comment:
Not sure to understand what this does, but it seems to always add those args, not only when there is ASM-only targets?

Reply:
This function generates the corresponding `/DEFAULTLIB:<CRT library>` options for the build type the user has specified. This is detailed in the tables here: [C runtime .lib files](https://learn.microsoft.com/en-us/cpp/c-runtime-library/crt-library-features?view=msvc-170#c-runtime-lib-files). 

It's possible that they're getting added for non NASM-only targets, because I couldn't understand the logic flow for the linking process and so I thought they ought to be added to the C part (as you noticed above).

Reply:
I've reviewed the code a bit deeper again and yes, they're inserted into all targets because of the `MSVCCompiler` mixin. This was added in #3981, see https://github.com/mesonbuild/meson/commit/fd2c3b4c77ac1977d254301876525f8e631a940a.

I believe the D compiler also uses a similar approach: https://github.com/mesonbuild/meson/blob/d0054f2c3c3497e22069d1efb5b1d985d75fe5ca/mesonbuild/compilers/d.py#L460-L487 https://github.com/mesonbuild/meson/blob/d0054f2c3c3497e22069d1efb5b1d985d75fe5ca/mesonbuild/compilers/d.py#L533-L538

Reply:
I've fixed this by ensuring NASM receives and redirects CRT queries to the host C compiler, like CUDA does.